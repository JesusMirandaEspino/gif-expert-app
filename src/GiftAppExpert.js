import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


const GiftAppExpert  = ( { defaultCategories = [] } ) => {

   // arreglo de categorias

    // no se declara asi const categorias = [ 'One Punch', 'Samurai X', 'Dragon Ball' ];
    const [ categories, setCategories ] = useState(  defaultCategories );

    /* Funcion de base
    const handleAdd = () => {
        setCategories( cat => [ ...cat,  'Demon Slayer'] );
    }
*/
    return (
        <section className="fila">
            <div className="contenedor1">

                <div className="col-full-12">
                    <h1> GiftAppExpert </h1>
                    <hr></hr>
                </div>

                <AddCategory setCategories={   setCategories  }/>

                <div className="col-full-12">
                    <ol>
                        {
                            categories.map( category => {
                                return <GifGrid 
                                        key={ category }
                                        category={ category }/>
                            } )
                        }
                    </ol>
                </div>

                <div className="col-full-12">
                </div>

            </div>
        </section>
    );
    
}

export default GiftAppExpert;