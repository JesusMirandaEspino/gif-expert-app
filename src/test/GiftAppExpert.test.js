import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import  GiftAppExpert  from '../GiftAppExpert';


describe( 'Pruebas  con componente <GiftAppExpert />', () => {


    test('Debe de mostrar el componente correctamente ', () => {

        const wrapper = shallow( <GiftAppExpert />  );

        expect( wrapper ).toMatchSnapshot();

    });


    test('Debe de mostrar una lista de categorias', () => {
        const categorias = [ 'one punch', 'dragon ball' ];
        const wrapper = shallow( <GiftAppExpert   defaultCategories={categorias}  />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'GifGrid' ).length ).toBe(categorias.length);

    });


});