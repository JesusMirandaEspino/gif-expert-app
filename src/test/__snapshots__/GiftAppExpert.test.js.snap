// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas  con componente <GiftAppExpert /> Debe de mostrar el componente correctamente  1`] = `
<section
  className="fila"
>
  <div
    className="contenedor1"
  >
    <div
      className="col-full-12"
    >
      <h1>
         GiftAppExpert 
      </h1>
      <hr />
    </div>
    <AddCategory
      setCategories={[Function]}
    />
    <div
      className="col-full-12"
    >
      <ol />
    </div>
    <div
      className="col-full-12"
    />
  </div>
</section>
`;

exports[`Pruebas  con componente <GiftAppExpert /> Debe de mostrar una lista de categorias 1`] = `
<section
  className="fila"
>
  <div
    className="contenedor1"
  >
    <div
      className="col-full-12"
    >
      <h1>
         GiftAppExpert 
      </h1>
      <hr />
    </div>
    <AddCategory
      setCategories={[Function]}
    />
    <div
      className="col-full-12"
    >
      <ol>
        <GifGrid
          category="one punch"
          key="one punch"
        />
        <GifGrid
          category="dragon ball"
          key="dragon ball"
        />
      </ol>
    </div>
    <div
      className="col-full-12"
    />
  </div>
</section>
`;
